ui/src/components/CsvSourceMapping.vue

```diff
@@ -65,6 +72,24 @@ export default Vue.extend({
     getColumnMappings () {
       return []
     },
+
+    async deleteSource (source: Source): Promise<void> {
+      this.$buefy.dialog.confirm({
+        title: source.actions.delete?.title,
+        message: 'Are you sure you want to delete this CSV source?',
+        confirmText: 'Delete',
+        type: 'is-danger',
+        hasIcon: true,
+        onConfirm: async () => {
+          const loading = this.$buefy.loading.open({})
+          try {
+            await this.$store.dispatch('cubeProjects/deleteSource', source)
+          } finally {
+            loading.close()
+          }
+        },
+      })
+    },
```

Comment:
What do you think about making `loading` and confirmation part of the state:

1. action `invoke operation` with the operation as parameter
   - if it's a `DELETE` request show a confirmation dialog
   - if it takes a body, display a form in the side panel
3. when user confirms/submits state the request a set a `loading` flag
4. when request finishes unset the `loading` flag